opnode/rollup/driver/state.go

```diff
@@ -173,19 +175,33 @@ func (s *state) findNextL1Origin(ctx context.Context) (eth.L1BlockRef, error) {
 	}
 	// Somehow reorg'd. Will let the state loop take care of it.
 	if curr.Hash != s.l2Head.L1Origin.Hash {
```

Comment:
Can cut this as we are now doing `BlockRefByHash` (was a holdover from when it was by number)